tracer/src/Datadog.Trace/Telemetry/Collectors/DependencyTelemetryCollector.cs

```diff
@@ -106,9 +98,18 @@ static bool IsBase32Char(char c)
             }
         }
 
-        private void SetHasChanges()
+        private static bool IsGuid(string assemblyName)
         {
-            Interlocked.Exchange(ref _hasChangesFlag, 1);
+            switch (assemblyName.Length)
+            {
+                // Simple implementation to remove guids
+                case 36 when assemblyName[8] == '-' && assemblyName[13] == '-' && assemblyName[18] == '-' && assemblyName[23] == '-':
+                // and other weird use cases like ℛ*710fa04a-6428-4dd1-85a0-0419c142709b#5-0
```

Comment:
It bugs me that we don't know where these are coming from 🤔 e.g. you've cater for the `#5-0` and `#13-0` suffix, but is `#123-0` possible? What about `#12345-0`? Given these massively blow up cardinality, maybe we should skip the upper bound, and just keep the `-` checks?